src/app/utils/helper.ts

```diff
@@ -0,0 +1,235 @@
+  const sectionData: any = dataToConsider.sections;
+
+  let headerGlobalCss: string = '';
+  let globalFontFamily: string = '';
+  let globalFontSize: string = '';
+
+  if (dataToConsider.styles.GLOBAL.backgroundColor) {
+    const backgroundColorJson = {
+      backgroundColor: dataToConsider.styles.GLOBAL.backgroundColor,
+    };
+
+    headerGlobalCss = await toCSS(JSON.stringify(backgroundColorJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontFamily) {
+    const fontFamilyJson = {
+      fontFamily: dataToConsider.styles.GLOBAL.fontFamily,
+    };
+
+    globalFontFamily = await toCSS(JSON.stringify(fontFamilyJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontSize) {
+    const fontSizeJson = {
+      fontSize: dataToConsider.styles.GLOBAL.fontSize,
+    };
+
+    globalFontSize = await toCSS(JSON.stringify(fontSizeJson));
+  }
+
+  let htmlData: string = '';
+  let headerHtml = '';
+  let firstColumnHtml: string[] = [];
+  let secondColumnHtml: string[] = [];
+  let imageUrl = '';
+
+  // iterate over the sections keys
+  const keys: string[] = Object.keys(sectionData);
+  keys.forEach((key: string) => {
+    const sectionDetails = sectionData[key];
+    const currSectioName = sectionDetails.sectionName;
+
+    const currSectionSchema = sectionMockJson.find(
+      (section) => section.name === currSectioName
+    );
+    if (!currSectionSchema) return;
+
+    // we also need to remove the <img tag from the titleComp as we'll be adding it via jspdf
+    currSectionSchema.titleComp = currSectionSchema.titleComp.replace(
+      /<img[^>]*>/g,
+      ''
+    );
+
+    // iterate over the keys to replace
+    allKeysToReplace.forEach((keyToReplace) => {
+      const keyToReplaceRegex = new RegExp('\\' + keyToReplace, 'g');
+
+      const keyToReplaceValue =
+        sectionDetails.body[0][keyToReplace.split('${this.')[1].split('}')[0]];
+
+      currSectionSchema.component = currSectionSchema.component.replace(
+        keyToReplaceRegex,
+        keyToReplaceValue
+      );
+
+      const keyToReplaceTitle =
+        sectionDetails.header[keyToReplace.split('${this.')[1].split('}')[0]];
+
+      if (keyToReplace === '${this.image}' && keyToReplaceTitle) {
+        imageUrl = keyToReplaceTitle;
+      }
+
+      currSectionSchema.titleComp = currSectionSchema.titleComp.replace(
+        keyToReplaceRegex,
+        keyToReplaceTitle
+      );
+    });
+
+    if (currSectionSchema.isStaticHeader) {
+      headerHtml += `
+        <div style="${headerGlobalCss}; padding: 15px 30px 15px 30px;">
+          ${currSectionSchema.titleComp}
+        </div>`;
+    }
+
+    if (!currSectionSchema.isStaticHeader) {
+      const sectionHtml = `
+      <div>
+        ${currSectionSchema.titleComp}
+        ${currSectionSchema.component}
+      </div>`;
+
+      if (sectionDetails.columnIndex === 0) {
+        firstColumnHtml[sectionDetails.positionIndex] = `
+          <div style="padding-bottom: 15px; padding-top: 15px;">
```

Comment:
If the layout of the page is full width (this should be ignored for 2 column layout), a section can have 2 different widths, this is dictated by a sections `activeWidth` value.
A section is either "full" or "half" for now. Half meaning half the width of the page. This can also mean that if 2 sections are next to each other and are both "half" width, they will fit in the same row

<img width="842" alt="image" src="https://github.com/dompainter/pdf-generator/assets/17257060/a470d72b-4395-4daa-bccf-9f54b7e43fee">

Reply:
Fixed.